LogForwarder/index.js

```diff
@@ -228,11 +230,17 @@ function parseData(logs, context) {
       return JSON.parse(logs); // for strings let's see if we can parse it into Object
     } catch {
       context.log.warn('cannot parse logs to JSON');
```

Comment:
Can we receive an array of plain text strings, each representing a log? In that case, this `context.log.warn()` would be noisy, wouldn't it?

Reply:
It could be an array of plain text strings. However, we are not doing JSON.parse() for every individual log lines. We try to do JSON.parse() for the logs param as a whole and in case of exception, the parsing wouldn't be successful. This means that the logs are not in perfect JSON format (since it is a plaint text) and we are safe to return what we receive.
For the question, since we do JSON.parse for the whole logs, the warning log message would occur only once!